### DISCUSSION TITLE
Hash check fails

### DISCUSSION THREAD

**Discussion by veljko** [ScoopInstaller/Scoop#6254] [Help and Questions]
Hi all
My pull request (https://github.com/ScoopInstaller/Extras/pull/14612) fails on hash check and I am not sure what is really going on. It looks like the old (5.7.1) and new (5.8.5) versions are mixed and hash is not correct.
Any help is appreciated,
Vel

**Comment by HUMORCE** [comment]
You're fixing a manifest that autoupdate is broken. You should execute `bin\checkver.ps1 -app <app>` after changes to confirm that `checkver` is working as expected, for `autoupdate` it's `bin\checkver.ps1 -app <app> -u`(or -f), `-u` = upgrade, `-f` force upgrade even it's already latest version(common used for hashes fix). the command lines should run at root dir of bucket. (i.e. ~\scoop\buckets\extras for default installation)

Hash extraction regex is wrong, because the hash information in HTML is:
```
<p>SHA256: <code>746ddade00b6a66a20b6f1378fcf4f6de2588f33be86285a5b8742c79d9363e5</code></p>
```

it should be `SHA256:\\s<code>$sha256`. `$sha256` is a built-in binding come from Scoop Core. See: https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#hash-variables
